Automattic/sensei
# Implement course import task

[course_import_test.txt](https://github.com/Automattic/sensei/files/4771418/course_import_test.txt)

### Overview
This PR implements the course import task.

- Every field should be imported correctly appart from prerequisites.
- Courses are compared by the slug to check if it is an existing course or a new one. If the course exists already, it is updated.
- The imported course status is 'draft' on course creation. The status is not updated on course update.
- Users that do not exist are created with a teacher role.

### Testing instructions
- To upload a CSV file you can navigate to Sensei LMS > Import. Once you upload the file click continue to start the import job.
- Make sure that the cronjobs have run.
- Alternatively you can run the job by using the CLI command `wp --user=<admin> --courses=<csv file> sensei-import`, which is a lot faster.
- Observe the imported courses.
- For an example CSV file you can see in the attached files.
